woocommerce/woocommerce-gutenberg-products-block
# Consistent HTML entities in Store API

This is an alternative fix to #1594 cc @haszari @nerrad 

Rather than strip entities for the term name, this adds a new formatting method to the API to be used on all HTML based strings which enforces entities for the response. This ensures all text/html coming from the API has the same formatting.

As well as formatting responses, I've included the `@wordpress/html-entities` package for decoding entities for display and included that for the attribute filter and all products block to fix #1586. 

As a note on usage, if using values from the API in react, you'll need to `decodeEntities` UNLESS you are using `dangerouslySetInnerHTML`, in which case entities will render correctly.

Fixes #1586 

### How to test the changes in this Pull Request:

1. Setup some attributes with & in the name, and some products with & in the title and/or excerpt.
2. Setup all products block. You can add 'summary' to the output to test excerpt handling.
3. Setup filter by attribute and active filter blocks.
4. View the all products block and confirm & rendering in the product title, summary, and filter blocks. 
5. Confirm filter blocks work, and active filters also shows correct & handling.
